kvlds/btree_find.c

```diff
@@ -155,11 +160,13 @@ findleaf(void * cookie)
 
 		/* Fetch the child node.  This adds a lock to the parent. */
 		if (btree_node_fetch(C->T, C->N, findleaf, C))
```

Comment:
I'm confused by this part.  "If the node is not present, fetch it".  So we fetch the node, then invoke `findleaf()` on it, pass it `C`... then is that instance of `findleaf()` supposed to free C?  If so, then the `btree_node_unlock(C->T, NP);` following the `btree_node_fetch()` is invalid, because `C->T` would then be de-allocating freed memory.

Reply:
`btree_node_fetch` requests the node to be paged in, but it returns before that operation completes; the callback happens later.  So in this case we're guaranteed that `findleaf(C)` has not been re-invoked at the time that `btree_node_fetch` returns.

Incidentally, this is why `findleaf` starts by descending *as far as possible* through the tree.  If we simply called a hypothetical `btree_descend` which went down one level, we would need to handle the cases of "the node was already paged in and we could descend immediately" vs. "the node needed to be retrieved and we descended in a callback function later" separately.